saige_assoc.py

```diff
@@ -0,0 +1,312 @@
+    default=output_path(
+        'nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1_cis_genePval'
+    ),
+)
+@click.option('--plink-path', default='/usr/local/bin/n.indep_100_n.cell_1_01.step1')
+@click.option('--gene-name', default='gene_1')
+@click.option('--chrom', default='2')
+@click.option('--cis-window-file', default='/usr/local/bin/gene_1_cis_region.txt')
+@click.option('--fit-null-mem', default='10Gi')
+@click.option('--run-sv-assoc-mem', default='10Gi')
+@click.option('--get-gene-pvals-mem', default='10Gi')
+def association_pipeline(
+    pheno_cov_filename: str,
+    vcf_file_path: str,
+    vcf_field: str,
+    covs_list: str,
+    sample_covs_list: str,
+    sample_id: str,
+    null_output_path: str,
+    sv_output_path: str,
+    gene_pvals_output_path: str,
+    plink_path: str,
+    gene_name: str,
+    chrom: str,
+    cis_window_file: str,
+    fit_null_mem: str,
+    run_sv_assoc_mem: str,
+    get_gene_pvals_mem: str,
+):
+    """
+    Run association for one gene
+    """
+
+    batch = get_batch('SAIGE-QTL pipeline')
+
+    # step 1 (fit null)
+    fit_null_job = batch.new_job(name='fit null')
+    fit_null_job.image(SAIGE_QTL_IMAGE)
+    fit_null_job.storage(fit_null_mem)
+    fit_null_job.declare_resource_group(
+        output={
+            'rda': '{root}.rda',
+            'varianceRatio': '{root}.varianceRatio.txt',
+        }
+    )
+    cmd = build_fit_null_command(
+        pheno_file=pheno_cov_filename,
+        cov_col_list=covs_list,
+        sample_cov_col_list=sample_covs_list,
+        sample_id_pheno=sample_id,
+        output_prefix=fit_null_job.output,
+        plink_path=plink_path,
+        pheno_col=gene_name,
+    )
+    fit_null_job.command(cmd)
+
+    # copy the output file to persistent storage?
+    if null_output_path.startswith('gs://'):
+        batch.write_output(fit_null_job.output, null_output_path)
+
+    # step 2 (cis eQTL single variant test)
+    run_sv_assoc_job = batch.new_job(name='single variant test')
+    run_sv_assoc_job.image(SAIGE_QTL_IMAGE)
+    run_sv_assoc_job.storage(run_sv_assoc_mem)
+    run_sv_assoc_job.depends_on(fit_null_job)
+    vcf_group = batch.read_input_group(vcf=vcf_file_path, index=f'{vcf_file_path}.csi')
+    cmd = build_run_single_variant_test_command(
+        vcf_file=vcf_group.vcf,
+        vcf_file_index=vcf_group.index,
+        vcf_field=vcf_field,
+        saige_output_file=run_sv_assoc_job.output,
+        chrom=chrom,
+        cis_window_file=cis_window_file,
+        gmmat_model_path=fit_null_job.output['rda'],
+        variance_ratio_path=fit_null_job.output['varianceRatio'],
+    )
+    run_sv_assoc_job.command(cmd)
+    if sv_output_path.startswith('gs://'):
+        batch.write_output(run_sv_assoc_job.output, sv_output_path)
+
+    # step3 (gene-level pvalues)
+    get_gene_pvals_job = batch.new_job(name='gene level pvalues')
+    get_gene_pvals_job.image(SAIGE_QTL_IMAGE)
+    get_gene_pvals_job.storage(get_gene_pvals_mem)
```

Comment:
A fussy little nit here, please be careful semantically about `mem` and `storage` - one is RAM, one is hard-drive capacity. The hail batch job commands `.memory()` and `.storage()` address very different needs, so this variable has a misleading name